test/qa/lib/firewall.py

```diff
@@ -93,98 +96,123 @@
+        f"-A INPUT -i {interface} -p {protocol} -m {protocol} --sport {port} -m comment --comment nordvpn -j ACCEPT",
+        f"-A INPUT -i {interface} -m connmark --mark 0xe1f1 -m comment --comment nordvpn -j ACCEPT",
+        f"-A INPUT -i {interface} -m comment --comment nordvpn -j DROP",
+        f"-A OUTPUT -o {interface} -p {protocol} -m {protocol} --dport {port} -m comment --comment nordvpn -j ACCEPT",
+        f"-A OUTPUT -o {interface} -p {protocol} -m {protocol} --sport {port} -m comment --comment nordvpn -j ACCEPT",
+        f"-A OUTPUT -o {interface} -m mark --mark 0xe1f1 -m comment --comment nordvpn -j CONNMARK --save-mark --nfmask 0xffffffff --ctmask 0xffffffff",
+        f"-A OUTPUT -o {interface} -m connmark --mark 0xe1f1 -m comment --comment nordvpn -j ACCEPT",
+        f"-A OUTPUT -o {interface} -m comment --comment nordvpn -j DROP"
+    ]
 
 
 def is_active(port="", protocol="", subnet=""):
-    # Get interface name of your default gateway
-    output = sh.ip.route.show("default")
-    print(sh.ip.route.show())
-    _, _, _, _, iface = output.split(None, 5)
+    """ returns True when all `expected_rules` are found in `current_rules` """
+    print(sh.ip.route())
+
+    default_route_interface = sh.ip.route.show("default").split(None)[4]
 
     try:
-        # Get VPN server's IP address
-        status = sh.grep(sh.nordvpn.status(), "IP")
-        _, server_ip = status.split(None, 2)
-    except sh.ErrorReturnCode:
+        server_ip = settings.get_server_ip()
+    except IndexError:
         server_ip = ""
-    print("Default gateway:", iface, "Server's IP:  ", server_ip)
 
-    rules = _get_firewall_rules(
-        daemon.is_killswitch_on(), server_ip, iface, port, protocol, subnet
+    print("Default gateway:", default_route_interface, "; Server IP:", server_ip)
+
+    expected_rules = _get_firewall_rules(
+        daemon.is_killswitch_on(),
+        server_ip,
+        default_route_interface,
+        port,
+        protocol,
+        subnet
     )
-    print("Expected rules:\n", rules)
+    print("Expected rules:\n")
+    for rule in expected_rules:
+        print(rule)
+    print("\n")
 
     current_rules = _get_iptables_rules()
-    print("Current rules:\n", current_rules)
+    print("Current rules:\n")
+    for rule in current_rules:
+        print(rule)
+    print("\n")
 
     print(sh.nordvpn.settings())
-    return rules in current_rules
+    return all(item in current_rules for item in expected_rules)
 
 
 def is_empty() -> bool:
     """returns True when firewall does not have DROP rules"""
     return "DROP" not in sh.sudo.iptables("-S")
 
 
-def _get_iptables_rules():
+def _get_iptables_rules() -> list[str]:
     # TODO: add full ipv6 support, separate task #LVPN-3684
     print("Using iptables")
-    return sh.sudo.iptables("-S")
+    return str(sh.sudo.iptables("-S")).split('\n')
+
+
+def add_route_delete_route():
```

Comment:
[minor] wouldn't it be better `add_and_delete_random_route` or `prepare_allow_list_tests`? :smile: